CONTRIBUTING.md

```diff
@@ -1,77 +1,312 @@
-# Contributing to convert2rhel
+# Contributing to Convert2RHEL
 
-## Coding guidelines
-1. All python code must be python 2.6/2.7/3.6 compatible
-1. The code should follow linting from pylint
+The following is a set of guidelines for contributing to Convert2RHEL codebase, which are hosted in
+the [OAMG Organization](https://github.com/oamg) on GitHub. These are mostly guidelines, not rules.
 
-## Developing locally
+### Table Of Contents
+
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+
+* [Working with GitHub](#working-with-github)
+* [Setting up Git](#setting-up-git)
+* [Forking a repository](#forking-a-repository)
+* [Collaborating with Pull Requests](#collaborating-with-pull-requests)
+
+[Getting started with development](#getting-started-with-development)
+
+* [Dependencies for local development](#dependencies-for-local-development)
+* [Setting up the environment](#setting-up-the-environment)
+* [Writing tests](#writing-tests)
+
+[Styleguide](#styleguide)
+
+* [Git commit messages](#git-commit-message)
+* [Python code styleguide](#python-code-styleguide)
+* [Python documentation styleguide](#python-documentation-styleguide)
+
+[Additional Notes](#additional-information)
+
+* [Building rpms locally](#building-rpms-locally)
+* [Building copr builds locally](#building-copr-builds-locally)
+
+## What should I know before I get started?
+
+Below are a list of things to keep in mind when developing and submitting contributions to this repository.
+
+1. All python code must be compatible with versions 2.6/2.7/3.6.
+2. The code should follow linting from pylint.
+3. All commits should have passed the pre-commit checks.
+4. Always squash your commits into a single one, [how to squash commits](https://stackoverflow.com/a/3825347).
```

Comment:
I've included a section to squash commits, since we do it in every pr.

Reply:
We don't necessarily need to squash commits if you have a good split between what each commit should do. Most of the time people squash to one commit as it makes it easier